python/apps/authentication/views.py

```diff
@@ -0,0 +1,33 @@
+from django.conf import settings
+from django.contrib.auth import get_user_model
+
+from google.auth.transport import requests
+from google.oauth2 import id_token
+from oauth2_provider.models import get_access_token_model
+from rest_framework import status
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+User = get_user_model()
+AccessToken = get_access_token_model()
+
+
+class AndroidAuthenticationView(APIView):
+    CLIENT_ID = settings.SOCIAL_AUTH_GOOGLE_OAUTH2_KEY
+
+    def get(self, request, *args, **kwargs):
+        try:
+            ID_token = self.request.query_params.get('id_token')
+            id_token_info = id_token.verify_oauth2_token(ID_token, requests.Request(), self.CLIENT_ID)
+
+            if id_token_info['iss'] not in ['accounts.google.com', 'https://accounts.google.com']:
+                raise ValueError('Wrong issuer.')
+
+            user_email = id_token_info['email']
+            user = User.objects.get(email=user_email)
```

Comment:
You do not handle a case when there is no such user.
Use `try` `except ObjectDoesNotExist` to catch that case